app/src/main/java/com/epam/themes/androidcomponents/AndroidComponentsActivity.java

```diff
@@ -0,0 +1,125 @@
+package com.epam.themes.androidcomponents;
+
+import android.content.ComponentName;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.ServiceConnection;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import com.epam.cleancodetest.R;
+import com.epam.themes.androidcomponents.activities.AActivity;
+import com.epam.themes.androidcomponents.activities.BActivity;
+import com.epam.themes.androidcomponents.activities.CActivity;
+
+public class AndroidComponentsActivity extends AppCompatActivity {
+
+    private Intent mIntent;
+    private Intent mBroadcastIntent = new Intent("com.epam.CUSTOM_ACTION");
+    private IntentFilter mIntentFilter = new IntentFilter("com.epam.CUSTOM_ACTION");
+    private ServiceConnection mServiceConnection = new SimpleServiceConnection();
+    private SimpleBroadcastReceiver mReceiver = new SimpleBroadcastReceiver();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_main);
+
+        setTitle(R.string.theme_android_components);
+
+        mIntent = new Intent(this, SimpleService.class);
+
+        getSupportFragmentManager()
+                .beginTransaction()
+                .add(R.id.fragmentView, new SimpleFragment())
```

Comment:
1. start app
2. press home
3. start app again - you will got 2 fragments here.